### DISCUSSION TITLE
MacBook Air M1 for AI

### DISCUSSION THREAD

**Discussion by omarabb315** [ml-explore/mlx#1472] [General]
is the MacBook Air M1 with 36 GB Unified-RAM Good for state-of-the-art finetuning ?

**Comment by awni** [comment]
I'm not sure what you mean by "state-of-the-art" fine-tuning.

If you are asking if you can fine-tune an LLM, yes definitely with a low-rank method and optionally a quantized model. You could for example fine-tune small to medium sized models (maybe up to 7b with QLoRA). It also depends on how big of a dataset you need to fine-tune on. If it's pretty small (like a few hundred samples, it should work fine and not take too long). 

Check out [fine-tuning in MLX LM](https://github.com/ml-explore/mlx-examples/blob/main/llms/mlx_lm/LORA.md) for more info on that.

**Comment by omarabb315** [comment]
Thanks a lot, yes I meant that finetuning tasks, 
Is there any benchmarks for its speed?
